src/main/java/qtlog/FilesystemController/FileMonitor.java

```diff
@@ -2,15 +2,35 @@
 
 import qtlog.DataModel.IFileObserver;
 import qtlog.shared.FileDTO;
-
+import qtlog.util.ConfigManager;
 import java.nio.file.WatchService;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.HashMap;
+import java.util.Map;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
 import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.StandardWatchEventKinds;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
 
-public class FileMonitor implements IFileMonitor, IFileObservable{
+public class FileMonitor implements IFileMonitor, IFileObservable, Runnable{
     IFileObserver observer;
     Path filepath;
     WatchService watchservice;
     FileDTO newestFile;
+    private boolean trace;
+    private Map<WatchKey, Path> keys;
+
+    public FileMonitor(){
+        this.trace = false;
+        this.filepath = Paths.get(ConfigManager.readLogPath());
+        this.newestFile = new FileDTO("dummy", "dummypath");
```

Comment:
The FileDTO class is pointless, I think we should use this instead:
```java
import java.nio.file.Path;
```

Reply:
Sure you could Path everywhere but I use the filename in some places as well which is helpful, I dont need to extract it from the path first, I rather not change every method where I use it for little gain. Also enables easy addition of more file information if we need it in the future

Reply:
(could add the Path data type to fileDTO tho if you want to use it)

Reply:
> I use the filename in some places as well which is helpful, I dont need to extract it from the path first
but Path has the method .getName which returns the filename. If you need to use it a bunch of times in a function you could assign the filename to a locally scoped variable.
> Also enables easy addition of more file information if we need it in the future
Actually I beg to differ, I believe Path gives us easy access to much more information and I don't see a reason to create extra work when it's already there in the standard library.
https://docs.oracle.com/javase/8/docs/api/java/nio/file/Path.html